internal/scan-manager/scanners/common/src/main/java/org/wso2/security/tools/scanmanager/scanners/common/ScannerController.java

```diff
@@ -71,16 +71,27 @@ public ResponseEntity startScan(@RequestBody ScannerScanRequest scannerScanReque
         ResponseEntity responseEntity;
         responseEntity = validateStartScanReq(scannerScanRequest);
         if (responseEntity.getStatusCode().equals(HttpStatus.ACCEPTED)) {
-            if (scanner.validateStartScan(scannerScanRequest)) {
-                log.info("Invoking start scan API.");
-                startScanThread = new Thread(() -> scanner.startScan(scannerScanRequest), "StartScanThread");
+
+            // If current scan request is to initiate new scan
+            if (!Boolean.parseBoolean(scannerScanRequest.getIsResume())) {
```

Comment:
It's more readable if we switch the two blocks IMO

Reply:
Resolved